feat: Add new option `defaultHoursMinutesValue`

AndrewLeeHc opened the pull request:
[why] Sometime user don't want 00:00 as default value after click td.day
[how] override this._setDate input value in td.day condition

OLC-1890

AndrewLeeHc committed:
feat: Add new option `defaultHoursMinutesValue`

[why] Sometime user don't want 00:00 as default value after click td.day
[how] override this._setDate input value in td.day condition

OLC-1890

Falldog commented on js/bootstrap-datetimepicker.js:
```diff
@@ -257,6 +257,35 @@
+      if (!defaultHoursMinutesValue) {
+        return 0;
+      }
+      return defaultHoursMinutesValue.hours;
```
why not convert to integer here? not outside

Falldog commented on js/bootstrap-datetimepicker.js:
```diff
@@ -992,6 +1021,11 @@
+                    // Override time if have default setting.
+                    const defaultHour = this.defaultHours();
+                    const defaultMinute = this.defaultMinutes();
+                    hours = Number.isInteger(defaultHour) ? defaultHour : hours;
+                    minutes = Number.isInteger(defaultMinute) ? defaultMinute : minutes;
```
please convert the defaultHour inside the defaultHours() function

AndrewLeeHc committed:
fix: Use `!== null` to check defaultHoursMinutesValue has value

[why] Number.isInteger's purpose is not clear
[how] return null if defaultHoursMinutesValue not defined

OLC-1890
